builtin/logical/jwt/path_roles.go

```diff
@@ -0,0 +1,197 @@
+			},
+			"key": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Private Key (RSA or EC) or String for HMAC Algorithm",
+			},
+			"lease": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     true,
+				Description: "Use leases, if true, tokens will be kept for 30 days, if false, tokens and claims are not stored",
+			},
+			"iss": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Default Issuer for the Role for the JWT Tokens",
+			},
+			"sub": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Default Subject for the Role for the JWT Token",
+			},
+			"aud": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Default Audience for the Role for the JWT Token",
+			},
+			"exp": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "0h",
+				Description: "Default Expiration (from time of issue) for the Role fpr the JWT Token, expressed in Duration, Example: 24h",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathRoleRead,
+			logical.WriteOperation:  b.pathRoleCreate,
+			logical.DeleteOperation: b.pathRoleDelete,
+		},
+
+		HelpSynopsis:    pathRolesHelpSyn,
+		HelpDescription: pathRolesHelpDesc,
+	}
+}
+
+func (b *backend) getRole(s logical.Storage, n string) (*roleEntry, error) {
+	entry, err := s.Get("role/" + n)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	var result roleEntry
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (b *backend) pathRoleDelete(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	err := req.Storage.Delete("role/" + data.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (b *backend) pathRoleRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	role, err := b.getRole(req.Storage, data.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return nil, nil
+	}
+
+	var r = structs.New(role).Map()
+
+	delete(r, "key")
+
+	resp := &logical.Response{
+		Data: r,
+	}
+
+	return resp, nil
+}
+
+func (b *backend) pathRoleCreate(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	name := data.Get("name").(string)
+	key  := data.Get("key").(string)
+	alg  := data.Get("algorithm").(string)
+	expRaw  := data.Get("exp").(string)
+	
+	signingMethod := jwt.GetSigningMethod(data.Get("algorithm").(string))
+	if signingMethod == nil {
+		return nil, errors.New("Invalid Signing Algorithm")
+	}
+
+	if key == "" {
+		return nil, errors.New("Key is Required")
+	}
+
+	if strings.HasPrefix(alg, "RS") {
+		// need RSA Private Key
+		if strings.Contains(key, "RSA PRIVATE KEY") == false {
```

Comment:
I would attempt to actually parse the key, since it may still be invalid. Same for HS and EC.